### PR TITLE
Deployments: Default strategy and strategy configuration

### PR DESCRIPTION
**Pull request by alvaroaleman**
**What this PR does / why we need it**:

Makes writing deployments easier as the strategy and strategyconfig gets defaulted

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
none
```

/assign @xrstf

### CODE REVIEW

File: api/pkg/resources/reconciling/wrapper_test.go [hunk 1]
```diff
@@ -531,3 +547,112 @@ func TestDefaultCronJob(t *testing.T) {
...
+		name   string
+		in     *appsv1.Deployment
+		verify func(*appsv1.Deployment) error
+	}{
+		{
+			name: "Strategy and strategysettings get defaulted",
+			in:   &appsv1.Deployment{},
+			verify: func(d *appsv1.Deployment) error {
+				if d.Spec.Strategy.Type != appsv1.RollingUpdateDeploymentStrategyType {
+					return fmt.Errorf("expectex strategy to be %q, was %q",
```
Comment (xrstf) [inline api/pkg/resources/reconciling/wrapper_test.go]: `Expected`

File: api/pkg/resources/reconciling/wrapper_test.go [hunk 2]
```diff
@@ -531,3 +547,112 @@ func TestDefaultCronJob(t *testing.T) {
...
+				}
+				if d.Spec.Strategy.RollingUpdate.MaxSurge.IntVal != 1 {
+					return fmt.Errorf("Expected .Spec.Strategy.RollingUpdate.MaxSurge to be 1, was %d",
+						d.Spec.Strategy.RollingUpdate.MaxSurge.IntVal)
+				}
+				if d.Spec.Strategy.RollingUpdate.MaxUnavailable == nil {
+					return errors.New("Expected .Spec.Strategy.RollingUpdate.MaxUnavailable to get defaulted, was nil")
+				}
+				if d.Spec.Strategy.RollingUpdate.MaxUnavailable.IntVal != 0 {
+					return fmt.Errorf("Expected Spec.Strategy.RollingUpdate.MaxUnavailable to be 0, was %d",
```
Comment (xrstf) [inline api/pkg/resources/reconciling/wrapper_test.go]: missing leading dot in front of .Spec

**Comment by alvaroaleman** [comment]
@xrstf Updated

File: api/pkg/resources/reconciling/wrapper_test.go [hunk 3]
```diff
@@ -531,3 +547,112 @@ func TestDefaultCronJob(t *testing.T) {
...
+		name   string
+		in     *appsv1.Deployment
+		verify func(*appsv1.Deployment) error
+	}{
+		{
+			name: "Strategy and strategysettings get defaulted",
+			in:   &appsv1.Deployment{},
+			verify: func(d *appsv1.Deployment) error {
+				if d.Spec.Strategy.Type != appsv1.RollingUpdateDeploymentStrategyType {
+					return fmt.Errorf("expected strategy to be %q, was %q",
```
Comment (xrstf) [inline api/pkg/resources/reconciling/wrapper_test.go:562]: Uppercase (though I get it, muscle memory, never create errors with uppercase letters, yadda yadda yadda).

Comment (alvaroaleman) [reply api/pkg/resources/reconciling/wrapper_test.go:562]: What?

**Pull request merged**